var/spack/repos/builtin/packages/hypre/package.py

```diff
@@ -93,7 +93,9 @@ class Hypre(AutotoolsPackage, CudaPackage):
     depends_on('superlu-dist', when='+superlu-dist+mpi')
 
     conflicts('+cuda', when='+int64')
-    conflicts('+unified-memory', when='~cuda')
+    conflicts('+rocm', when='+int64')
+    conflicts('+rocm', when='@:2.23.0')
```

Comment:
HIP support added in v2.21.0
```suggestion
    conflicts('+rocm', when='@:2.20')
```